aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java

```diff
@@ -423,4 +539,66 @@ private void syncHostTemperatureAndHumidySensor(boolean fullSync) {
       }
    }
 
+   public Map<String,Map<String,String>> getMetricsNameAndSensorsIdMap(Map<String,Map<String,String>> metricsNameAndSensorsMap,
+         Asset sensor){
+      String positionInfo = null;
+      int position = sensor.getCabinetUnitPosition();
+      if(position == 0) {
+         positionInfo = FlowgateConstant.DEFAULT_CABINET_UNIT_POSITION;
+      }
+      positionInfo = String.valueOf(position);
+      switch (sensor.getMountingSide()) {
+      case Front:
+         switch (sensor.getSubCategory()) {
+         case Temperature:
+            Map<String,String> frontTempMap  = metricsNameAndSensorsMap.get(MetricName.SERVER_FRONT_TEMPERATURE);
+            if(frontTempMap == null) {
+               frontTempMap = new HashMap<String,String>();
+            }
+            frontTempMap.put(sensor.getId(), positionInfo);
+            metricsNameAndSensorsMap.put(MetricName.SERVER_FRONT_TEMPERATURE, frontTempMap);
+            break;
+         case Humidity:
+            Map<String,String> frontHumidityMap  = metricsNameAndSensorsMap.get(MetricName.SERVER_FRONT_HUMIDITY);
+            if(frontHumidityMap == null) {
+               frontHumidityMap = new HashMap<String,String>();
+            }
+            frontHumidityMap.put(sensor.getId(), positionInfo);
+            metricsNameAndSensorsMap.put(MetricName.SERVER_FRONT_HUMIDITY, frontHumidityMap);
+            break;
+         default:
+            break;
+         }
+         break;
+      case Back:
+         switch (sensor.getSubCategory()) {
+         case Temperature:
+            Map<String,String> backTempMap  = metricsNameAndSensorsMap.get(MetricName.SERVER_BACK_TEMPREATURE);
+            if(backTempMap == null) {
+               backTempMap = new HashMap<String,String>();
+            }
+            backTempMap.put(sensor.getId(), positionInfo);
+            metricsNameAndSensorsMap.put(MetricName.SERVER_BACK_TEMPREATURE, backTempMap);
+            break;
+         case Humidity:
+            Map<String,String> backHumidityMap  = metricsNameAndSensorsMap.get(MetricName.SERVER_BACK_HUMIDITY);
+            if(backHumidityMap == null) {
+               backHumidityMap = new HashMap<String,String>();
+            }
+            backHumidityMap.put(sensor.getId(), positionInfo);
+            metricsNameAndSensorsMap.put(MetricName.SERVER_BACK_HUMIDITY, backHumidityMap);
+            break;
+         default:
+            break;
+         }
+         break;
+      case External:
+      case Unmounted:
+         break;
+      default:
```

Comment:
what if the sensor mounting side is not front or back?  

We should consider if the front/back formula don't have value, then we should use this sensor for it. Otherwise we can ignore it.

Reply:
Done